cmake/leatherman.cmake

```diff
@@ -147,3 +167,19 @@ function(get_commit_string varname)
         set(${varname} " (commit ${GIT_SHA1})" PARENT_SCOPE)
     endif()
 endfunction()
+
+include(GenerateExportHeader)
+# Usage: symbol_exports(TARGET HEADER)
+#
+# Generate the export header for restricting symbols exported from the library,
+# and configure the compiler. Restricting symbols has several advantages, noted
+# at https://gcc.gnu.org/wiki/Visibility.
+macro(symbol_exports target header)
+    generate_export_header(${target} EXPORT_FILE_NAME "${header}")
+    if (NOT APPLE)
```

Comment:
Why is this "NOT APPLE"? Do we also need to constrain output stuff on other unices besides linux (IIRC AIX doesn't support visibility?)